### PR TITLE
Revert grdecl

### PR DESCRIPTION
**Pull request by joakim-hove**
**Issue**
Revert offending commit from https://github.com/equinor/ert/issues/1423

**Approach**
"git revert" + rewrite testdata files to ensure they are long enough for the file-type detection algorithm

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

**Comment by ManInFez** [comment]
Test this please!

File: tests/res/enkf/test_field_config.py [hunk 1]
```diff
@@ -30,9 +30,8 @@ def test_field_guess_filetype(self):
                 f.write("-- my comment\n")
                 f.write("-- more comments\n")
                 f.write("SOWCR\n")
-                for i in range(128 // 8):
+                for i in range(256 // 8):  # technicalities demand file has >= 256B
```
Comment (jondequinor) [inline tests/res/enkf/test_field_config.py]: Could you expand on this? What technicalities

Comment (joakim-hove) [reply tests/res/enkf/test_field_config.py]: This commit is old - I tried (not hard enough) to eradicate it, bringing it back in now.

The point is that deep down there is some code which need to know whether en eclipse file should be opened as ASCII or binary, that code uses some heuristics to guess whether the file is binary or not. The heuristic is based on checking bit 8 - and if bit 8 in the bytestream is set in sufficiently many bytes the files is assumed to be binary -  now this heuristic needs a filesize before it willing to make a guess. This code is soon old enough to vote and should probably be taken out.

Comment (jondequinor) [reply tests/res/enkf/test_field_config.py]: Okay, thank you. Can you add this explanation to the comment? This comment is going to disappear, the comment will add value.

Also, why do we have to guess? Can't the user inform us in some way? The way I see it, guessing on binary data is a complete gamble.

Comment (joakim-hove) [reply tests/res/enkf/test_field_config.py]: > Can you add this explanation to the comment?

> Also, why do we have to guess?

Don't have to of course - but that is the way it is implemented right now. Typical situation is the `INIT_FILES:` attribute of a field configuration:
```
FIELD PORO PARAMETER poro.grdecl INIT_FILES:fields/poro%d
```
The files in the `INIT_FILES:` attribute can be of the types: 
1. Eclipse grdecl file
2. ASCII Eclipse restart file
3. BInary Eclipse restart file
4. RMS roff file

And that file type is determined/guessed based on file content - with no attention to naming conventions.

Comment (jondequinor) [reply tests/res/enkf/test_field_config.py]: Great, thank you 👌🏻

**Comment by ManInFez** [comment]
Test this please!

**Review by ManInFez — APPROVED** [review]

File: tests/res/enkf/test_field_config.py (hunk 1 above)
Comment (lars-petter-hauge) [reply tests/res/enkf/test_field_config.py]: Edit: never mind, I misunderstood where the filename came from, the argument to `INIT_FILES` doesn't include the extension then. Leaving the comment here though just to have possible extensions listed.

Or.. you could of course check the extension of the file given by the argument, don't need the user to tell us more than the basename.

So the possible files can have the following extensions:

1. `GRID`, `FGRID`, `EGRID` and `FEGRID` (Reflects: BINARY / ASCII FORMATTED / BINARY EXTENDED/ ASCII FORMATTED EXTENDED)
2. `UNRST` and `Xnnnn`  (Reflects: BINARY / BINAR (per step))
3.  `FUNRST` and `Fnnnn` (Reflects:  FORMATTED / FORMATTED (per step))
4. `ROFF`

Not sure if I've forgotten any?

Is there a reason why we couldn't check on the specific extensions?

@joakim-hove 

I wouldn't be surprised if there was a reason for this being implemented in a different manner.

**Pull request merged**